### ISSUE TITLE
link custom collection to advanced source search page

### ISSUE THREAD

**Issue by rahulbot** [mediacloud/web-tools#1638]
People will want to see the list of sources matching their custom collections - we should link the custom collection widget to the advanced source search results page on Source Mgr. This will let people page through the matching sources (and ideally let them download that list as well).   

![Search___Explorer___Media_Cloud](https://user-images.githubusercontent.com/673178/62792544-24dd2900-baed-11e9-8142-5ae2da52b8d9.jpg)

I'm pretty sure this is blocked by #1637, because we can't encode a custom collection over to that page in a URL (and it doesn't support multiple selections within a metadata filter).

**Comment by cindyloo** [comment]
here is another Custom Collection implementation issue....

**Comment by rahulbot** [comment]
Yep. My thought here seems to have been to (on #1637) we use some kind of wrapper that parses the URL-encoded "custom collection criteria" and then passes that in as `initialValues` to the media-picker-based advanced search page.

**Comment by cindyloo** [comment]
within MediaPicker,  upon the clicking of `Add as Custom Collection` we need to do two things:
- store all the media_ids that are part of this selection
- show them all as selected
Then, when the "OK" button is clicked, we serialize those media_ids into the URL as we do for sources and collections, along with the `media_keyword` and the metadata tags. That way, we can utilize the custom collection mechanism outside of the MediaPicker

**Comment by rahulbot** [comment]
Can you explain that approach more? I like how you architected it when you first implement it; having a "custom collection" just be a list of the media metadata tags the user selected (ie. the search criteria). Why do we need to turn that into a list of media to support using "custom collections" in other places?

I think this particular issue, which probably isn't high priority, would entail:
1. Adding a click handler to the custom collection rendering
2. When clicked, finding a way to serialize the selected media metadata search into a URL and appending that to the link over to advanced search page
3. Parsing that serialized data on the advanced search page
4. Passing the parsed "custom collection" into the media picker as preselected search criteria for the drop down menus

**Comment by cindyloo** [comment]
right, we don't want or need to serialize all media_ids into the URL - we want to special case the search + media_keyword + metadata tag in python to insert those media ids into the solr search query when needed eg `media_id: {[...]}`.
we do want to display correlated media as selected in MediaPicker display page

**Comment by cindyloo** [comment]
fyi the Custom Collection handling is currently broken, which makes this a more timely consideration

**Comment by rahulbot** [comment]
Ah. I hear two considerations here:

**1)** How do we search in Explorer with a custom collection?
I don't think we need to expand to media ids in the solr search query for custom collections. We can search these at `tags_id_media` parameters - I think the code in [`concatenate_query_for_solr` does this already](https://github.com/mitmedialab/MediaCloud-Web-Tools/blob/443a6c8609c2879d51c722ae647f673a6267cfd1/server/views/media_picker.py#L182). Am I misunderstanding your point?

**2)** How do we show matching media as "selected" in the media picker sources search results?
This feels like a nice-to-have feature, but is a great thought to mention that I totally overlooked. The list of media on that screen are pulled from a regular `mediaList` call, so they include all the media metadata tags on them for each source. With that in mind, what I'd do is implement this idea as a check in JS that evaluates each media source returned against any "custom collections" criteria to see if it matches. If it matches any of them, or is selected as an individual media source, then we can mark is as selected. That logic should be to hard, because it is just code that would compare `tags_id`s (respecting the AND'ing between metadata categories, and OR'ing between items within the same metadata category).

**Comment by cindyloo** [comment]
1. yes, this is correct, I was reading the code to mean custom_collection_ids, but actually we never passed in the collection_ids, but rather the metadata tags_media_ids. That said,  handling of custom collections and multiple custom collections had been broken
this case for example:
'[{"media_keyword": "atlanta", "tags_id_media": "[]"},{"media_keyword": "atlanta", "tags_id_media": "[[9353663]]"},{"media_keyword": "", "tags_id_media": "[[9353663]]"}]'
should properly translate to
(atlanta) OR (atlanta AND tags_id_media: (9353663)) OR (* AND tags_id_media: (9353663))
but we aren't handling multiple custom collections nor were we allowing for the case when we have a keyword but no metadata tags

**Comment by cindyloo** [comment]
proper translation I believe is
`'(*) AND (( tags_id_media:(58722749)  OR \'atlanta\' OR \'atlanta\' AND (tags_id_media:(9353663)) ) OR \'search\' AND (tags_id_media:(9353663)) ))'`

**Comment by cindyloo** [comment]
we have removed the keyword from the custom collection as there is no way to persist this keyword - no way to run a solr query with (keyword and metadata tags). The keyword is merely used to search for sources within mediapicker.  Hence, the above query is reduced to '(obama) AND (( tags_id_media:(58722749) OR (tags_id_media:(9353663)) ) OR  (tags_id_media:(9353663)) ))'

**Comment by cindyloo** [comment]
remember the Source Manager expects custom collections in this form:
`tags=%7B"1935":%5B9353498%5D%7D`
such that the full url should look something like this:
`http://localhost:5000/#/search?search=bbc&tags=%7B%221935%22:%5B9353498%5D%7D`
